### ISSUE TITLE
Shift of a negative value in listfdb (master, libming 0.4.8 and earlier)

### ISSUE THREAD

**Issue by hlef** [libming/libming#106]
The `readSBits` function (util/listfdb.c) in libming through 0.4.8 is vulnerable to a shift of a negative value. This issue also affects the master branch.

Reproduce:
`$ listfdb negative-left-shift-in-listfdb-poc-1.fdb`

Output:
```
Font Name: f
number of glyphs: 1

off 0, 12,0
Offset0: -8
Glyph 0:
listfdb.c:115:14: runtime error: shift exponent -1 is negative
```

You can find the reproducer [here](https://github.com/hlef/pocs/blob/master/negative-left-shift-in-listfdb-poc-1.fdb).

**Comment by hlef** [comment]
This issue is identical to #97 because the code itself is identical (very likely to be duplicate code). Other (now fixed) issues in util/read.c are almost certainly still affecting util/listfdb.c.

I will investigate them, report the bugs for clarity and submit PRs.

I'm not quite sure this is worth asking for CVE numbers, I'll decide after a closer look.

**Comment by hlef** [comment]
Fixed in c4d20b1.
